bool pin_1
bool pin_2
bool pin_3
bool pin_4
bool[] pins